### ISSUE TITLE
[regression] 8bitdo controllers (SN30 pro and pro+) have wrong button mappings in 2.28

### ISSUE THREAD

**Issue by kelvie** [libsdl-org/SDL#7925]
Both 8btido controllers are in Xbox mode.

I downgraded to 2.26, and it seems to work again, I also posted on the arch forums: https://bbs.archlinux.org/viewtopic.php?id=286919

I think this has something to do with SDL now reading from hidraw directly? This is only over bluetooth. Wired over USB, it works completely fine (pressing select shows BTN_SELECT in evtest, over bluetooth it shows up as BTN_TL).

Basically the exact same mapping seen in this bug: https://github.com/atar-axis/xpadneo/issues/291

Let me know what debug information you guys need.

**Comment by Squall-Leonhart** [comment]
See https://github.com/libsdl-org/SDL/issues/7524

**Comment by kelvie** [comment]
I don't see how that's related, they're different controllers (and the symptoms are different).

**Comment by slouken** [comment]
Okay, I know what's going on here. I removed the Xbox controller mappings in favor of the automatic Linux mapping code, but this controller isn't recognized by the kernel, so the driver falls back to thinking it's the original Xbox controller with BTN_C and BTN_Z.

I need to restore some of the Bluetooth mappings, but I need to do more investigation to figure out which ones.

**Comment by slouken** [comment]
Fixed! :)

**Comment by kelvie** [comment]
Super appreciate it!
